diff --git a/package.json b/package.json
index e0f267a..1d58662 100644
--- a/package.json
+++ b/package.json
@@ -19,7 +19,7 @@
     "morgan": "^1.10.0",
     "openai": "^4.77.0",
     "path": "^0.12.7",
-    "youtube-dl-exec": "3.0.15",
+    "youtube-dl-exec": "3.0.10",
     "yt-scraper": "^2.2.0",
     "ytdl-core": "^4.11.5"
   },
diff --git a/src/app.js b/src/app.js
index c2a2438..e5abbc8 100644
--- a/src/app.js
+++ b/src/app.js
@@ -65,11 +65,11 @@ app.post('/analyze-video', async (req, res) => {
     return res.send("error: invalid_domain")
   }
 
-  const duration = await getVideoDuration(urlVideo)
+  // const duration = await getVideoDuration(urlVideo)
 
-  if (duration === null) {
-    return res.send("video longer than 10 minutes");
-  }
+  // if (duration === null) {
+  //   return res.send("video longer than 10 minutes");
+  // }
 
   // Diretório temporário
   const tempDir = path.resolve(path.resolve(), 'src/downloads');
@@ -84,16 +84,15 @@ app.post('/analyze-video', async (req, res) => {
     // Baixa o áudio do vídeo do YouTube
     // A função `youtubedl` é utilizada para executar o download do áudio do vídeo fornecido na URL (`urlVideo`).
     // O áudio será salvo no caminho especificado (`audioPath`) no formato de melhor qualidade disponível (`bestaudio`).
-    youtubedl(urlVideo, {
+    await youtubedl(urlVideo, {
       output: audioPath,
-      format: 'bestaudio',
-      extractAudio: true,
-      audioFormat: 'mp3',
-      cookies: cookiesFile // Se necessário
-    }).then(output => {
-      console.log('Áudio baixado com sucesso!', output);
+      format: 'bestaudio'
+    }).then(() => {
+      console.log("Áudio baixado com sucesso:", audioPath);
     }).catch(err => {
-      console.error('Erro ao baixar áudio:', err);
+      console.error("Erro ao baixar áudio:", err);
+      isProcessing = false; // Certifique-se de liberar o estado
+      throw err;
     });
 
 
diff --git a/yarn.lock b/yarn.lock
index 5fd3254..7a5a8b9 100644
--- a/yarn.lock
+++ b/yarn.lock
@@ -1383,10 +1383,10 @@ yallist@^4.0.0:
   resolved "https://registry.yarnpkg.com/yallist/-/yallist-4.0.0.tgz#9bb92790d9c0effec63be73519e11a35019a3a72"
   integrity sha512-3wdGidZyq5PB084XLES5TpOSRA3wjXAlIWMhum2kRcv/41Sn2emQ0dycQW4uZXLejwKvg6EsvbdlVL+FYEct7A==
 
-youtube-dl-exec@3.0.15:
-  version "3.0.15"
-  resolved "https://registry.yarnpkg.com/youtube-dl-exec/-/youtube-dl-exec-3.0.15.tgz#c155ef1322f6acf6ad5d3a0ad30478f14ed70541"
-  integrity sha512-QVXOxeUSeID8UzE+HmQ5TN7xDMf0xI22MUslb3n/jTHTd8uXw2F9wbCR+I34HFZcNKY6qxTcmDy6REbJAMPing==
+youtube-dl-exec@3.0.10:
+  version "3.0.10"
+  resolved "https://registry.yarnpkg.com/youtube-dl-exec/-/youtube-dl-exec-3.0.10.tgz#89f5fada997572b112dcf84ae93f6981e16cd7e8"
+  integrity sha512-t3ih+3bn2rFYSStuVjKVHUPyPYhPvPjIPjJZAzjFb6qD8uJxgJ5GHicSwbPkezM8IVdnoKPRkZ6XuIPHCqRRZg==
   dependencies:
     bin-version-check "~6.0.0"
     dargs "~7.0.0"
