diff --git a/Dockerfile b/Dockerfile
index 45d4238..014f743 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -17,7 +17,7 @@ RUN yarn install --frozen-lockfile
 COPY . /app/
 
 # Expor a porta 3000
-EXPOSE 3000
+EXPOSE 8080
 
 # Rodar o servidor (ajuste se necessário)
 CMD ["node", "src/index.js"]
diff --git a/src/app.js b/src/app.js
index 3dfc84f..0d53140 100644
--- a/src/app.js
+++ b/src/app.js
@@ -72,7 +72,7 @@ app.post('/analyze-video', async (req, res) => {
   // }
 
   // Diretório temporário
-  const tempDir = path.resolve(path.resolve(), 'src/downloads');
+  const tempDir = path.resolve(path.resolve(), 'app/src/downloads');
   if (!fs.existsSync(tempDir)) fs.mkdirSync(tempDir, { recursive: true });
 
   // Caminhos temporários únicos
diff --git a/src/transcribeAudioToText.js b/src/transcribeAudioToText.js
index f789cbb..8157d55 100644
--- a/src/transcribeAudioToText.js
+++ b/src/transcribeAudioToText.js
@@ -9,7 +9,7 @@ if (process.env.NODE_ENV !== "production") {
 const openai = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });
 
 export default async function transcribeAudioToText() {
-  const filePath = path.join(path.resolve(), "src/downloads/audio.mp3");
+  const filePath = path.join(path.resolve(), "app/src/downloads/audio.mp3");
   console.log(filePath)
 
   const audioStream = fs.createReadStream(filePath);
